plugins/net_plugin/net_plugin.cpp

```diff
@@ -3428,10 +3470,53 @@ namespace eosio {
       }
    }
 
+   // called from any thread
+   void net_plugin_impl::update_security_group(const block_state_ptr& bs) {
+      // update cache
+      //    - connection_mtx is needed
+      //
+      auto& update = bs->get_security_group_info();
+      std::shared_lock<std::shared_mutex> connection_guard(connections_mtx);
+      if(!security_group.update_cache(update.version, update.participants)) {
+         return;
+      }
+
+      std::vector<connection_ptr> added_connections;
+      added_connections.reserve(connections.size());
+
+      // update connections
+      //    - connection_mtx still needed
+      //
+      for(auto& connection : connections) {
+         const auto& participant = connection->participant_name();
+         if(!participant) {
+            return;
+         }
+         if(security_group.is_in_security_group(participant.value())) {
```

Comment:
Need to change sec;urity_group_manager::is_in_security_group to always return true if cache_.empty().  This will allow this logic here to work if we ever remove all participants in the security group and it allows Dmytro's code to only have to call that method to determine what he should set participating_ to.

Reply:
Modified is_in_security_group to also return true when the cache_ is empty